src/query/expression/src/evaluator.rs

```diff
@@ -89,6 +104,33 @@ impl<'a> Evaluator<'a> {
         }
     }
 
+    fn cache_expr_value(&self, expr: &Expr, value: Value<AnyType>) {
+        let mut cached_values_ref = self.cached_values.as_ref().unwrap().write();
+        if let Entry::Vacant(v) = cached_values_ref.entry(expr.clone()) {
+            v.insert(value);
+        }
+    }
+
+    fn get_cached_expr_value(&self, expr: &Expr) -> Option<Value<AnyType>> {
```

Comment:
toooo complex, this function is read overhead

Reply:
if let Some(cache_values) = self.cached_values
                .as_ref() {
                
         if let Entry::Occupied(v) = cached_values_ref.entry(expr.clone()) {
             return Some(v);
        }
   }